smithy-protocol-test-traits/src/test/resources/software/amazon/smithy/protocoltests/traits/errorfiles/all-malformed-request-features.smithy

```diff
@@ -0,0 +1,69 @@
+namespace smithy.example
+
+use smithy.test#httpMalformedRequestTests
+
+@trait
+@protocolDefinition
+structure testProtocol {}
+
+@trait
+@authDefinition
+structure testScheme {}
+
+@http(method: "POST", uri: "/")
+@httpMalformedRequestTests([
+    {
+        id: "bodyRegex",
+        documentation: "Testing...",
+        protocol: testProtocol,
+        request: {
+            body: "Hi",
+            headers: {"X-Foo": "baz"},
+            host: "example.com",
+            method: "POST",
+            uri: "/",
+            queryParams: ["foo=baz"]
+        },
+        response: {
+            code: 400,
+            headers: {"X-Foo": "baz"},
+            bodyMessageRegex: "Invalid JSON: .*",
+            bodyMediaType: "application/json"
+        },
+        tags: ["foo", "bar"]
+    },
+    {
+        id: "body",
+        documentation: "Testing...",
+        protocol: testProtocol,
+        request: {
+            body: "Hi",
+            headers: {"X-Foo": "baz"},
+            host: "example.com",
+            method: "POST",
+            uri: "/",
+            queryParams: ["foo=baz"]
+        },
+        response: {
+            code: 400,
+            headers: {"X-Foo": "baz"},
+            body: """
+            {
+                "message" : "Invalid JSON"
```

Comment:
Is this the same test as above but checking body v/s bodyMessageRegex? The regex has ": .*" but it doesn't here...

Reply:
you can either specify body (exact match) or a regex for the message. this test is simply "can we deserialize all of these fields"